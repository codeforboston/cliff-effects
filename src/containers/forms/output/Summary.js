import { connect } from 'react-redux';

import { Summary } from '../../../forms/output/Summary';

const mapStateToProps  = (state) => {
  return { client: state.get('client') };
};

export default connect(mapStateToProps)(Summary);
