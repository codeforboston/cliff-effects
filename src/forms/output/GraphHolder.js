import React, { Component } from 'react';

// CUSTOM COMPONENTS
import { GraphTimeButtons } from '../../components/GraphTimeButtons';


class GraphHolder extends Component {

  constructor (props) {
    super(props);
    this.state = { activeID: 'Yearly' };
  }

  onClick = (evnt) => {
    let id = evnt.target.id;
    this.setState({ activeID: id });
  };

  render () {
    const { activeID }  = this.state,
          { Graph, client }         = this.props,
          { current }               = client,
          // The ids later used to access all program-specific data and functions
          // Only active programs are added
          activePrograms            = [ ...current.benefits ];

    return (
      <div className='graph-holder'>
        <Graph
          className='client-graph'
          client={ client }
          timescale={ activeID }
          activePrograms={ activePrograms } />
        <GraphTimeButtons
          activeID={ activeID }
          onClick={ this.onClick } />
      </div>
    );
  };  // End render()

};  // End <GraphHolder>


export { GraphHolder };
