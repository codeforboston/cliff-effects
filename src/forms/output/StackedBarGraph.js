import React from 'react';
import { Bar } from 'react-chartjs-2';

// COMPONENT HELPER FUNCTIONS
import {
  formatAxis,
  formatLabel,
  formatStackedTitle,
} from '../../utils/charts/chartFormatting';
import { applyAndPushBenefits } from '../../programs/applyAndPushBenefits';

// DATA
// Colors and text for parts of the chart
import { PROGRAM_CHART_VALUES } from '../../utils/charts/PROGRAM_CHART_VALUES';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


/** Visual representation of the table
 *
 * Alternatives:
 * 1. Stacked area graph for two data points
 * 2. Line graph of difference of just total money coming in
 * 
 * @param {object} client Full client object, with current
 *     and future. All client props are needed.
 */
const StackedBarGraph = function({ client }) {

  var clone = cloneDeep(client),
      curr  = clone.current;

  var allData         = {},
      activeBenefits  = [ `income` ];

  if (curr.hasSection8) {
    activeBenefits.push(`section8`);
  }

  if (curr.hasSnap) {
    activeBenefits.push(`snap`);
  }

  var currentCalcData = {
    activeBenefits: activeBenefits,
    dataToAddTo:    allData,
    clientToChange: clone,
    timeframe:      `current`,
  };
  applyAndPushBenefits (currentCalcData);

  // Add to the `current` data already there
  var futureCalcData = {
    activeBenefits: activeBenefits,
    dataToAddTo:    allData,
    clientToChange: clone,
    timeframe:      `future`,
  };
  applyAndPushBenefits (futureCalcData);

  var datasets    = [],
      moneyLabels = [];
  for (let bName of activeBenefits) {

    let frosting = PROGRAM_CHART_VALUES[ bName ],
        dataset  = {
          label:           frosting.name,
          backgroundColor: frosting.color,
          data:            allData[ bName ],
        };

    if (bName === `income`) {
      dataset.fill = `origin`;
      for (let amount of dataset.data) {
        moneyLabels.push(Math.round(amount));
      }
    }

    datasets.push(dataset);
  }  // end for each benefit in order

  const stackedBarProps = {
    data: {
      labels:   moneyLabels,
      datasets: datasets,
    },
    options: {
      title: {
        display: true,
        text:    'Money Coming In as Income Changes',
      },
      scales: {
        yAxes: [
          {
            stacked:    true,
            scaleLabel: {
              display:     true,
              labelString: 'Total Money Coming In ($)',
            },
            ticks: {
              beginAtZero: true,
              callback:    formatAxis,
            },
          }, 
        ],
        xAxes: [
          {
            stacked:    true,
            scaleLabel: {
              display:     true,
              labelString: 'Monthly Income ($)',
            },
            ticks: { callback: formatAxis },
          },
        ],
      },
      tooltips: {
        callbacks: {
          title: formatStackedTitle,
          label: formatLabel,
        },
      },
    },
  };


  return (
    <Bar { ...stackedBarProps } />
  );

};  // End <StackedBarGraph>


export { StackedBarGraph };
