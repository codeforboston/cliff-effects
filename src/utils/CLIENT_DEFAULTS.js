import { cloneDeep } from 'lodash';

/**
 * Default client values.
 * MUST ALWAYS BE VALID
 * 
 * Note: A 'positive number' money value can have more than
 *     two decimal places.
 */
const CLIENT_DEFAULTS = {

  // Property Name                | Default Value   | Valid Value Description

  current: {
    // Current programs
    hasSnap:                        false,            // boolean
    hasHousing:                     false,            // boolean
    // Household
    // 'm_' for 'member'
    household:  [{                  // 'head' member  // one or more member objects
        m_age:                      30,               // whole number
        m_role:                     'head',           // 'head', 'spouse', 'member'
        m_disabled:                 false             // boolean
    }],
    // Income
    earned:                         0,                // positive number (can have more than two decimal places)
    TAFDC:                          0,                // positive number
    SSI:                            0,                // positive number
    SSDI:                           0,                // positive number
    childSupportIn:                 0,                // positive number
    unemployment:                   0,                // positive number
    workersComp:                    0,                // positive number
    pension:                        0,                // positive number
    socialSecurity:                 0,                // positive number
    alimony:                        0,                // positive number
    otherIncome:                    0,                // positive number
    incomeExclusions:               0,                // positive number
    // Expenses
    childDirectCare:                0,                // positive number
    childBeforeAndAfterSchoolCare:  0,                // positive number
    childTransportation:            0,                // positive number
    childOtherCare:                 0,                // positive number
    earnedBecauseOfChildCare:       0,                // positive number
    childSupportPaidOut:            0,                // positive number
    adultDirectCare:                0,                // positive number
    adultTransportation:            0,                // positive number
    adultOtherCare:                 0,                // positive number
    disabledAssistance:             0,                // positive number
    earnedBecauseOfAdultCare:       0,                // positive number
    disabledMedical:                0,                // positive number
    otherMedical:                   0,                // positive number
    /** @todo When client has section 8, switch this to 'housingVoucher' */
    /** @todo Make shelter expense values more robust. */
    shelter:                        'homeless',       // 'homeless', 'homeowner', 'renter', 'housingVoucher'
    contractRent:                   0,                // positive number
    rentShare:                      0,                // positive number
    rent:                           0,                // positive number
    mortgage:                       0,                // positive number
    housingInsurance:               0,                // positive number
    propertyTax:                    0,                // positive number
    climateControl:                 false,            // boolean
    nonHeatElectricity:             false,            // boolean
    phone:                          false,            // boolean
    fuelAssistance:                 false,            // boolean  Note: Input is 'yes', 'no', but should be being converted
    otherExpenses:                  0                 // positive number
  },

  // Note: A `future` value will default to the client's `current` value until a user changes it explicitly
  future: {},                       // same structure // same

};

CLIENT_DEFAULTS.future = cloneDeep( CLIENT_DEFAULTS.current );


export { CLIENT_DEFAULTS };
