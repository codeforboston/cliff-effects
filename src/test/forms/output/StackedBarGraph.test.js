import React from 'react';
import { mount } from 'enzyme';
import { cloneDeep, set } from 'lodash';

import { StackedBarGraph } from '../../../forms/output/StackedBarGraph';
import { CLIENT_DEFAULTS } from '../../../utils/CLIENT_DEFAULTS';

jest.mock('react-chartjs-2', () => {
  const BarMock = () => {
    return null;
  };

  return {
    Bar(props) {
      return <BarMock { ...props } />;
    },
  };
});

describe('<StackedBarGraph>', () => {
  let client;

  const buildGraph = () => {
    return mount(<StackedBarGraph client={ client } />);
  };

  beforeEach(() => {
    client = cloneDeep(CLIENT_DEFAULTS);
  });

  it('renders with snap and current earned less than future earned', () => {
    set(client, 'current.hasSnap', true);
    set(client, 'current.earned', 100);
    set(client, 'future.earned', 200);

    expect(buildGraph()).toMatchSnapshot();
  });

  it('renders with snap and current earned greater than future earned', () => {
    set(client, 'current.hasSnap', true);
    set(client, 'current.earned', 200);
    set(client, 'future.earned', 100);
    
    expect(buildGraph()).toMatchSnapshot();
  });

  it('renders with section8', () => {
    set(client, 'current.hasSection8', true);
    set(client, 'current.earned', 100);
    set(client, 'future.earned', 200);

    expect(buildGraph()).toMatchSnapshot();
  });

  it('renders with both snap and section8', () => {
    set(client, 'current.hasSnap', true);
    set(client, 'current.hasSection8', true);
    set(client, 'current.earned', 100);
    set(client, 'future.earned', 200);

    expect(buildGraph()).toMatchSnapshot();
  });
});
