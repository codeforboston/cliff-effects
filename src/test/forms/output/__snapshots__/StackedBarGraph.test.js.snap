// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StackedBarGraph> renders with both snap and section8 1`] = `
<StackedBarGraph
  client={
    Object {
      "current": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 100,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": true,
        "hasSnap": true,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
      "future": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 200,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": false,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
    }
  }
>
  <Bar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgb(136, 187, 236)",
            "data": Array [
              100,
              200,
            ],
            "fill": "origin",
            "label": "Income",
          },
          Object {
            "backgroundColor": "rgba(206, 203, 61, 1)",
            "data": Array [
              0,
              0,
            ],
            "label": "Section 8 Housing",
          },
          Object {
            "backgroundColor": "rgba(101, 47, 138, 1)",
            "data": Array [
              192,
              192,
            ],
            "label": "SNAP",
          },
        ],
        "labels": Array [
          100,
          200,
        ],
      }
    }
    options={
      Object {
        "scales": Object {
          "xAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Monthly Income ($)",
              },
              "stacked": true,
              "ticks": Object {
                "callback": [Function],
              },
            },
          ],
          "yAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Total Money Coming In ($)",
              },
              "stacked": true,
              "ticks": Object {
                "beginAtZero": true,
                "callback": [Function],
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "Money Coming In as Income Changes",
        },
        "tooltips": Object {
          "callbacks": Object {
            "label": [Function],
            "title": [Function],
          },
        },
      }
    }
  >
    <BarMock
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgb(136, 187, 236)",
              "data": Array [
                100,
                200,
              ],
              "fill": "origin",
              "label": "Income",
            },
            Object {
              "backgroundColor": "rgba(206, 203, 61, 1)",
              "data": Array [
                0,
                0,
              ],
              "label": "Section 8 Housing",
            },
            Object {
              "backgroundColor": "rgba(101, 47, 138, 1)",
              "data": Array [
                192,
                192,
              ],
              "label": "SNAP",
            },
          ],
          "labels": Array [
            100,
            200,
          ],
        }
      }
      options={
        Object {
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Monthly Income ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "callback": [Function],
                },
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Total Money Coming In ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "beginAtZero": true,
                  "callback": [Function],
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "text": "Money Coming In as Income Changes",
          },
          "tooltips": Object {
            "callbacks": Object {
              "label": [Function],
              "title": [Function],
            },
          },
        }
      }
    />
  </Bar>
</StackedBarGraph>
`;

exports[`<StackedBarGraph> renders with section8 1`] = `
<StackedBarGraph
  client={
    Object {
      "current": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 100,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": true,
        "hasSnap": false,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
      "future": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 200,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": false,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
    }
  }
>
  <Bar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgb(136, 187, 236)",
            "data": Array [
              100,
              200,
            ],
            "fill": "origin",
            "label": "Income",
          },
          Object {
            "backgroundColor": "rgba(206, 203, 61, 1)",
            "data": Array [
              0,
              0,
            ],
            "label": "Section 8 Housing",
          },
        ],
        "labels": Array [
          100,
          200,
        ],
      }
    }
    options={
      Object {
        "scales": Object {
          "xAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Monthly Income ($)",
              },
              "stacked": true,
              "ticks": Object {
                "callback": [Function],
              },
            },
          ],
          "yAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Total Money Coming In ($)",
              },
              "stacked": true,
              "ticks": Object {
                "beginAtZero": true,
                "callback": [Function],
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "Money Coming In as Income Changes",
        },
        "tooltips": Object {
          "callbacks": Object {
            "label": [Function],
            "title": [Function],
          },
        },
      }
    }
  >
    <BarMock
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgb(136, 187, 236)",
              "data": Array [
                100,
                200,
              ],
              "fill": "origin",
              "label": "Income",
            },
            Object {
              "backgroundColor": "rgba(206, 203, 61, 1)",
              "data": Array [
                0,
                0,
              ],
              "label": "Section 8 Housing",
            },
          ],
          "labels": Array [
            100,
            200,
          ],
        }
      }
      options={
        Object {
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Monthly Income ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "callback": [Function],
                },
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Total Money Coming In ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "beginAtZero": true,
                  "callback": [Function],
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "text": "Money Coming In as Income Changes",
          },
          "tooltips": Object {
            "callbacks": Object {
              "label": [Function],
              "title": [Function],
            },
          },
        }
      }
    />
  </Bar>
</StackedBarGraph>
`;

exports[`<StackedBarGraph> renders with snap and current earned greater than future earned 1`] = `
<StackedBarGraph
  client={
    Object {
      "current": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 200,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": true,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
      "future": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 100,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": false,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
    }
  }
>
  <Bar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgb(136, 187, 236)",
            "data": Array [
              200,
              100,
            ],
            "fill": "origin",
            "label": "Income",
          },
          Object {
            "backgroundColor": "rgba(101, 47, 138, 1)",
            "data": Array [
              192,
              192,
            ],
            "label": "SNAP",
          },
        ],
        "labels": Array [
          200,
          100,
        ],
      }
    }
    options={
      Object {
        "scales": Object {
          "xAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Monthly Income ($)",
              },
              "stacked": true,
              "ticks": Object {
                "callback": [Function],
              },
            },
          ],
          "yAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Total Money Coming In ($)",
              },
              "stacked": true,
              "ticks": Object {
                "beginAtZero": true,
                "callback": [Function],
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "Money Coming In as Income Changes",
        },
        "tooltips": Object {
          "callbacks": Object {
            "label": [Function],
            "title": [Function],
          },
        },
      }
    }
  >
    <BarMock
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgb(136, 187, 236)",
              "data": Array [
                200,
                100,
              ],
              "fill": "origin",
              "label": "Income",
            },
            Object {
              "backgroundColor": "rgba(101, 47, 138, 1)",
              "data": Array [
                192,
                192,
              ],
              "label": "SNAP",
            },
          ],
          "labels": Array [
            200,
            100,
          ],
        }
      }
      options={
        Object {
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Monthly Income ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "callback": [Function],
                },
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Total Money Coming In ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "beginAtZero": true,
                  "callback": [Function],
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "text": "Money Coming In as Income Changes",
          },
          "tooltips": Object {
            "callbacks": Object {
              "label": [Function],
              "title": [Function],
            },
          },
        }
      }
    />
  </Bar>
</StackedBarGraph>
`;

exports[`<StackedBarGraph> renders with snap and current earned less than future earned 1`] = `
<StackedBarGraph
  client={
    Object {
      "current": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 100,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": true,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
      "future": Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 200,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": false,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      },
    }
  }
>
  <Bar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgb(136, 187, 236)",
            "data": Array [
              100,
              200,
            ],
            "fill": "origin",
            "label": "Income",
          },
          Object {
            "backgroundColor": "rgba(101, 47, 138, 1)",
            "data": Array [
              192,
              192,
            ],
            "label": "SNAP",
          },
        ],
        "labels": Array [
          100,
          200,
        ],
      }
    }
    options={
      Object {
        "scales": Object {
          "xAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Monthly Income ($)",
              },
              "stacked": true,
              "ticks": Object {
                "callback": [Function],
              },
            },
          ],
          "yAxes": Array [
            Object {
              "scaleLabel": Object {
                "display": true,
                "labelString": "Total Money Coming In ($)",
              },
              "stacked": true,
              "ticks": Object {
                "beginAtZero": true,
                "callback": [Function],
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "Money Coming In as Income Changes",
        },
        "tooltips": Object {
          "callbacks": Object {
            "label": [Function],
            "title": [Function],
          },
        },
      }
    }
  >
    <BarMock
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgb(136, 187, 236)",
              "data": Array [
                100,
                200,
              ],
              "fill": "origin",
              "label": "Income",
            },
            Object {
              "backgroundColor": "rgba(101, 47, 138, 1)",
              "data": Array [
                192,
                192,
              ],
              "label": "SNAP",
            },
          ],
          "labels": Array [
            100,
            200,
          ],
        }
      }
      options={
        Object {
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Monthly Income ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "callback": [Function],
                },
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Total Money Coming In ($)",
                },
                "stacked": true,
                "ticks": Object {
                  "beginAtZero": true,
                  "callback": [Function],
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "text": "Money Coming In as Income Changes",
          },
          "tooltips": Object {
            "callbacks": Object {
              "label": [Function],
              "title": [Function],
            },
          },
        }
      }
    />
  </Bar>
</StackedBarGraph>
`;
